controllers/item.js

```diff
@@ -122,27 +128,72 @@ item.withCustomFields = (req, queryBuilder) => {
       this.on('customField.customFieldID', 'itemCustomField.customFieldID')
         .on('item.barcode', 'itemCustomField.barcode')
     })
+    .modify(item.withFieldType.bind(null, req))
     .where('item.barcode', req.params.barcode)
 }
-item.getCustomFields = endpoint.getAll('customField', {
-  modify: item.withCustomFields
-})
-item.getCustomField = endpoint.get('itemCustomField', 'customFieldID', {
-  modify: item.withCustomFieldDetails,
-  hasOrganizationID: false
-})
+
+// Field type table for local reference
+const fieldType = {
+  text: 1,
+  number: 2,
+  currency: 3
+}
+
+// Convert value to correct type based on field type
+const typeCastValue = (fieldTypeID, value) => {
+  if (fieldTypeID === fieldType.number) {
+    return Number.parseInt(value, 10)
+  } else if (fieldTypeID === fieldType.currency) {
+    return Number.parseFloat(value)
```

Comment:
Would it be worth making sure the value here has two digits?

Reply:
Also, if dealing with digits is too much of a hassle, something that we do at work is to convert all currencies `* 100`. This way, you can still send a number without having to worry about the digits. Then the front end just has to `/ 100` to display the correct value.

Reply:
Hm, I'm not sure I understand. What would that tell us about the data if we know it has two digits?

Reply:
Oh maybe I don't understand what this does correctly. Is it meant to format the data sent *to* the client, or format the data stored in the db sent *from* the client?

Reply:
This function is run before a response is sent, so it formats the data before it gets sent to the client. All it actually does is convert a string to the correct type. Does that help?

Reply:
Oh ok, so I just wanted to make sure we display currencies with two digits on the app, so it could be formatted on either the client or the api (it also might not be worth sending a float with many digits when we only need 2). Is that enforced when storing values to the database?

Reply:
The backend casts it to a string on the way into the database, so it should preserve all digits, then on the way back it should have exactly the same number of digits. That seems like a presentational constraint rather than a data constraint - I think that you should enforce the number of digits on the front-end, either by validating the input or or formatting the response from the backend. What do you think?

Reply:
Ok yeah that makes sense